test/e2e/vsphere/machines.go

```diff
@@ -0,0 +1,288 @@
+package vsphere
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/api/machine/v1beta1"
+	machinesetclient "github.com/openshift/client-go/machine/clientset/versioned/typed/machine/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/openshift/machine-api-operator/pkg/controller/vsphere"
+	util "github.com/openshift/machine-api-operator/test/e2e"
+)
+
+const (
+	machineRole         = "e2e-test"
+	machineReadyTimeout = time.Minute * 6
+)
+
+type MachineTest struct {
```

Comment:
Need this be exported?

Reply:
These probably can be private/internal structs.  Its really just for the machine/machineset tests.  I was thinking of making a base struct that stores the common parts of these two types.

Reply:
I have removed these and refactored the test code.